src/multicore.rs

```diff
@@ -1,81 +1,129 @@
 //! An interface for dealing with the kinds of parallel computations involved in
-//! `bellman`. It's currently just a thin wrapper around [`CpuPool`] and
-//! [`crossbeam`] but may be extended in the future to allow for various
-//! parallelism strategies.
-//!
-//! [`CpuPool`]: futures_cpupool::CpuPool
+//! `bellman`. It's currently just a thin wrapper around [`rayon`] but may be
+//! extended in the future to allow for various parallelism strategies.
 
 #[cfg(feature = "multicore")]
 mod implementation {
-    use crossbeam::{self, thread::Scope};
-    use futures::{Future, IntoFuture, Poll};
-    use futures_cpupool::{CpuFuture, CpuPool};
+    use std::env;
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    use crossbeam_channel::{bounded, Receiver};
+    use lazy_static::lazy_static;
+    use log::{error, trace};
     use num_cpus;
 
-    #[derive(Clone)]
-    pub struct Worker {
-        cpus: usize,
-        pool: CpuPool,
+    static WORKER_SPAWN_COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+    lazy_static! {
+        static ref NUM_CPUS: usize = env::var("BELLMAN_NUM_CPUS")
+            .map_err(|_| ())
+            .and_then(|num| num.parse().map_err(|_| ()))
+            .unwrap_or_else(|_| num_cpus::get());
+        // See Worker::compute below for a description of this.
+        static ref WORKER_SPAWN_MAX_COUNT: usize = *NUM_CPUS * 4;
+        pub static ref THREAD_POOL: rayon::ThreadPool = rayon::ThreadPoolBuilder::new()
+            .num_threads(*NUM_CPUS)
+            .build()
+            .unwrap();
     }
 
-    impl Worker {
-        // We don't expose this outside the library so that
-        // all `Worker` instances have the same number of
-        // CPUs configured.
-        pub(crate) fn new_with_cpus(cpus: usize) -> Worker {
-            Worker {
-                cpus,
-                pool: CpuPool::new(cpus),
-            }
-        }
+    #[derive(Clone)]
+    pub struct Worker {}
 
+    impl Worker {
         pub fn new() -> Worker {
-            Self::new_with_cpus(num_cpus::get())
+            Worker {}
         }
```

Comment:
Either add a `Default` implementation or suppress the clippy lint.

Reply:
Clippy lints will be addressed separately.